go/provider/client/client.go

```diff
@@ -106,45 +106,38 @@ type reqClient struct {
 // provider queries, the provider's address, and optional ClientOption functions for customizing
 // the client configuration.
 //
-// The following options can be provided.
+// The following options can be provided:
 //   - WithAuthCerts: Configure TLS certificates for secure communication
 //   - WithAuthJWTSigner: Set a JWT signer for authentication
 //   - WithAuthToken: Provide an authentication token
+//   - WithQueryClient: Set a query client for on-chain provider discovery (for on-chain operations)
+//   - WithProviderURL: Set a provider URL directly (for off-chain operations)
 //
-// Note, auth have the following priority: WithAuthCerts > WithAuthJWTSigner > WithAuthToken
+// Note: WithQueryClient and WithProviderURL are mutually exclusive.
+// Auth options have the following priority: WithAuthCerts > WithAuthJWTSigner > WithAuthToken
 //
 // The function will:
 // 1. Apply any provided ClientOptions
-// 2. Query the provider's host URI using the QueryClient
+// 2. Query the provider's host URI using the QueryClient (if provided) or use the direct URL
 // 3. Set up TLS configuration with system certificates
 // 4. Configure client authentication using either provided certificates or JWT signing
 //
 // Returns an error if:
 // - Any ClientOption fails to apply
-// - The provider query fails
+// - Both WithQueryClient and WithProviderURL are provided
+// - Neither WithQueryClient nor WithProviderURL are provided
+// - The provider query fails (when using QueryClient)
 // - The host URI is invalid
 // - System certificates cannot be loaded
-func NewClient(ctx context.Context, qclient aclient.QueryClient, addr sdk.Address, opts ...ClientOption) (Client, error) {
-	res, err := qclient.Provider(ctx, &ptypes.QueryProviderRequest{Owner: addr.String()})
-	if err != nil {
-		return nil, err
-	}
-
-	uri, err := url.Parse(res.Provider.HostURI)
-	if err != nil {
-		return nil, err
-	}
-
+func NewClient(ctx context.Context, addr sdk.Address, opts ...ClientOption) (Client, error) {
```

Comment:
Gave it another thought, and, terribly I think we can make it better.
Provider client should not need to worry about blockchain at all, only parameter it should receive is `provider-url`, and it is caller's responsibility to figure out where this url comes from.

Lets remove `qclient` from here completely and then let `cli` and other clients to process `url` detection:
- for cli it will be `provider-url` flag or blockchain
- some other clients may opt to cache provider url in own unique way (surely they can do with current options as well, however, it would be better to keep blockchain parts separate)